components/sup/src/manager/service/mod.rs

```diff
@@ -926,10 +927,24 @@ impl Service {
     fn run_health_check_hook(&mut self) {
         let _timer = hook_timer("health-check");
         debug!("Running Health Check hook for ({})", self.spec_ident);
+        // We'd like to capture how long it takes for a health check
+        // to run for eventing purposes. We're already keeping a time
+        // for Prometheus (see `_timer`, above), but we can't inspect
+        // that and get the elapsed time. Thus, we keep track of time
+        // ourselves.
+        //
+        // Additionally, we'll only send the time with the health
+        // check event if there's actually a hook that runs. We'll
+        // default to there not being a hook, though.
+        let mut execution_duration = None;
```

Comment:
I find the value of avoiding `mut` pretty great since it makes it far more likely that the compiler will catch a mistake when the code is changed in the future. We can avoid it if we refactor a tiny bit:
```rust
        let (check_result, execution_duration) = if let Some(ref hook) = self.hooks.health_check {
            …
            (result, Some(event_start.elapsed()))
        } else {
            (match self.supervisor.status() {
                 …
             },
             None)
        };
```

Reply:
Excellent suggestion